### ISSUE TITLE
Crash when viewing Ponder

### ISSUE THREAD

**Issue by NillerMedDild** [Create-Additions/Ponderjs#1]
A few Enigmatica 6 players reported when leaving our Tiny Dry Rubber Ponder, so... here's the crash log 😄 

https://gist.github.com/NillerMedDild/d1bd5abf154dfb14c5bcbce56d1237a2

**Comment by kotakotik22** [comment]
the tile at the given location doesn't exist at the current time, this could be because you are creating the tile at the same time as you are trying to modify the nbt

**Comment by kotakotik22** [comment]
i will make it throw an exception explaining the error without crashing though

**Comment by kotakotik22** [comment]
seems like i was right, these: 
https://github.com/NillerMedDild/Enigmatica6/blob/4a04264214719daae1aa217e2de75f6d303dc9db/kubejs/client_scripts/ponder/tech/industrialforegoing/latex.js#L102-L112
https://github.com/NillerMedDild/Enigmatica6/blob/4a04264214719daae1aa217e2de75f6d303dc9db/kubejs/client_scripts/ponder/tech/industrialforegoing/latex.js#L114-L124
create blocks at the same time as they try to modify their NBT. i suggest adding ``scene.idle(1)`` to idle 1 tick before doing that

**Comment by kotakotik22** [comment]
you should check this
https://github.com/Creators-of-Create/Create/wiki/Internal---Ponder-UI#blocking-and-non-blocking-instructions
it will explain why this is happening, if you dont understand

**Comment by NillerMedDild** [comment]
I will do just that, thanks a lot for the help and explaination :D

**Comment by Quezler** [comment]
if these lines are removed the `laser_drill` ponder doesn't crash right after the last keyframe on the third page:

https://github.com/NillerMedDild/Enigmatica6/blob/7ca03b3aff0443f4837d646d1fb664c7b2705bc5/kubejs/client_scripts/ponder/tech/industrialforegoing/laser_drill.js#L420-L426

~~but in the `./logs/kubejs<client/server/startup>.txt` are no traces of anything gone wrong, in fact it doesn't save any of the 3 when it crashes: (timestamp is 1-2 minutes behind)~~

^ written when i still had my debug version installed, then i recompiled the normal one, then forgot to add it to mods so i had to reboot yet again, and now here we are 30 minutes later, but even with the code unmodified its still behind & has no mention:

[client.txt](https://github.com/Create-Additions/Ponderjs/files/7444033/client.txt)
[server.txt](https://github.com/Create-Additions/Ponderjs/files/7444034/server.txt)
[startup.txt](https://github.com/Create-Additions/Ponderjs/files/7444035/startup.txt)

**Comment by kotakotik22** [comment]
just to confirm, you are using the latest ponderjs version (1.0.3b), right?

**Comment by kotakotik22** [comment]
okay so these lines throw the exception:
https://github.com/Create-Additions/Ponderjs/blob/39e937073d1d2673c4dea3c6b6c050b63558bc3a/src/main/java/com/kotakotik/ponderjs/util/SceneBuilderJS.java#L135-L138
but im not sure why it doesnt show the correct message

this is not an issue with the coordinates, but something wrong with the passed in MapJS

**Comment by kotakotik22** [comment]
nevermind, sorry, misread my own code lol

the problem is that it cant turn the nbt of the block at that position into a MapJS

**Comment by kotakotik22** [comment]
i dont really know why this could happen, so this would probably be a better question to kubejs devs

**Comment by Quezler** [comment]
will you be forwarding the issue to them or should we?

**Comment by Quezler** [comment]
> the problem is that it cant turn the nbt of the block at that position into a MapJS

its trying to map the wrong block though (an energy cell instead of the fluid tank), or does the code try to map all the nbt blocks in the scene first before trying to edit just one?

**Comment by kotakotik22** [comment]
oh, sorry, missed that, then its not a kubejs issue

**Comment by kotakotik22** [comment]
i think you might be debugging the wrong line, ill try debugging myself

**Comment by Quezler** [comment]
> oh, sorry, missed that, then its not a kubejs issue

might still be both, it fails to map the mnt of the wrong block, so that block might still be broken even when it tries to edit the right block

**Comment by NillerMedDild** [comment]
I think this might just be a case of kubejs converting the number in the nbt to double. Will test it this evening.

**Comment by Quezler** [comment]
dumped a bunch of `System.out.println` in the kubejs code, and this is what i have found:

this is where the nbt enters kubejs:
https://github.com/KubeJS-Mods/KubeJS/blob/main/common/src/main/java/dev/latvian/kubejs/util/MapJS.java#L28

then in this method it gets returned early by one of the `||`'s in the very first if:
https://github.com/KubeJS-Mods/KubeJS/blob/main/common/src/main/java/dev/latvian/kubejs/util/UtilsJS.java#L240

then on this line the `.getClass().getName()`of `o1` is `net.minecraft.nbt.CompoundNBT`:
https://github.com/KubeJS-Mods/KubeJS/blob/main/common/src/main/java/dev/latvian/kubejs/util/MapJS.java#L29

here is the output, keep in mind that the lines are slightly off from the above ones since the println shifts them:

```
[10:11:10] [Render thread/INFO] [STDOUT/]: [dev.latvian.kubejs.util.UtilsJS:wrap:240]: {RSControl:{RSMode:0b,RSPower:0b,RSThreshold:0b},Facing:3b,ItemInv:[{Slot:0b,id:"thermal:rf_coil_creative_augment",Count:1b}],Enchantments:[],AmountOut:1000,Security:{SecUUID:"1ef1a6f0-87bc-4e78-0a0b-c6824eb787ea",SecAccess:0b,SecName:"[None]"},Active:0b,Xfer:{XferOut:1b,XferIn:0b},Energy:16000000,x:3,y:2,z:1,id:"thermal:energy_cell",Sides:[B;2B,2B,2B,2B,2B,2B],AmountIn:1000}
[10:11:10] [Render thread/INFO] [STDOUT/]: [dev.latvian.kubejs.util.MapJS:of:29]: {RSControl:{RSMode:0b,RSPower:0b,RSThreshold:0b},Facing:3b,ItemInv:[{Slot:0b,id:"thermal:rf_coil_creative_augment",Count:1b}],Enchantments:[],AmountOut:1000,Security:{SecUUID:"1ef1a6f0-87bc-4e78-0a0b-c6824eb787ea",SecAccess:0b,SecName:"[None]"},Active:0b,Xfer:{XferOut:1b,XferIn:0b},Energy:16000000,x:3,y:2,z:1,id:"thermal:energy_cell",Sides:[B;2B,2B,2B,2B,2B,2B],AmountIn:1000}
[10:11:10] [Render thread/INFO] [STDOUT/]: [dev.latvian.kubejs.util.MapJS:of:30]: net.minecraft.nbt.CompoundNBT
```

but for context here are screenshots of the lines of code responsible for the above output:
![Screen Shot 2021-10-30 at 10 21 01](https://user-images.githubusercontent.com/3179271/139525745-4c2cfed5-a4db-484f-bc7b-04ace8aa1b2f.png)
![Screen Shot 2021-10-30 at 10 21 10](https://user-images.githubusercontent.com/3179271/139525747-5c15b052-09ee-4ce5-99ad-96fb7a3b90cb.png)

**Comment by Quezler** [comment]
i've tried replacing the thermal energy cells with powah energy cells in the nbt file, but it kept crashing so it doesn't seem to be caused by _just_ the thermal energy cell.

currently i have commented out the last two lines from `mapConsumerToNBT` in order to not modify anything but still see all the attempts that are being made:

```
[12:33:17] [Render thread/INFO] [STDOUT/]: [com.kotakotik.ponderjs.util.SceneBuilderJS$WorldInstructionsJS:lambda$mapJsConsumerToNBT$0:135]: new mapJsConsumerToNBT
[12:33:17] [Render thread/INFO] [STDOUT/]: [dev.latvian.kubejs.util.MapJS:of:31]: {redstone_mode:0,energy_stored_main_energy:9000000000000000000L,side_transfer_type:[I;1,1,1,1,1,1],Size:2,x:3,variant:7,y:2,z:1,Items:[],id:"powah:energy_cell",energy_capacity_main_energy:9000000000000000000L}
[12:33:17] [Render thread/INFO] [STDOUT/]: [dev.latvian.kubejs.util.MapJS:of:32]: net.minecraft.nbt.CompoundNBT
[12:33:17] [Render thread/INFO] [STDOUT/]: [com.kotakotik.ponderjs.util.SceneBuilderJS$WorldInstructionsJS:lambda$mapJsConsumerToNBT$0:135]: new mapJsConsumerToNBT
[12:33:17] [Render thread/INFO] [STDOUT/]: [dev.latvian.kubejs.util.MapJS:of:31]: {pointer:0,redstoneManager:{LastState:0b,Name:"ignore"},energyStorage:{energy:0},x:3,y:3,workingBar:{MaxProgress:50,Tick:50},z:1,showingArea:0b,id:"industrialforegoing:laser_drill",augmentInventory:{Size:4,FacingModes:{LEFT:"NONE",TOP:"NONE",BACK:"NONE",FRONT:"NONE",RIGHT:"NONE",BOTTOM:"NONE"},Items:[]},target:-824633692101L}
[12:33:17] [Render thread/INFO] [STDOUT/]: [dev.latvian.kubejs.util.MapJS:of:32]: net.minecraft.nbt.CompoundNBT
[12:33:17] [Render thread/INFO] [STDOUT/]: [com.kotakotik.ponderjs.util.SceneBuilderJS$WorldInstructionsJS:lambda$mapJsConsumerToNBT$0:135]: new mapJsConsumerToNBT
[12:33:17] [Render thread/INFO] [STDOUT/]: [dev.latvian.kubejs.util.MapJS:of:31]: {redstone_mode:0,energy_stored_main_energy:9000000000000000000L,side_transfer_type:[I;1,1,1,1,1,1],Size:2,x:1,variant:7,y:2,z:3,Items:[],id:"powah:energy_cell",energy_capacity_main_energy:9000000000000000000L}
[12:33:17] [Render thread/INFO] [STDOUT/]: [dev.latvian.kubejs.util.MapJS:of:32]: net.minecraft.nbt.CompoundNBT
[12:33:17] [Render thread/INFO] [STDOUT/]: [com.kotakotik.ponderjs.util.SceneBuilderJS$WorldInstructionsJS:lambda$mapJsConsumerToNBT$0:135]: new mapJsConsumerToNBT
[12:33:17] [Render thread/INFO] [STDOUT/]: [dev.latvian.kubejs.util.MapJS:of:31]: {pointer:0,redstoneManager:{LastState:0b,Name:"ignore"},energyStorage:{energy:0},x:1,y:3,workingBar:{MaxProgress:50,Tick:50},z:3,showingArea:0b,id:"industrialforegoing:laser_drill",augmentInventory:{Size:4,FacingModes:{LEFT:"NONE",TOP:"NONE",BACK:"NONE",FRONT:"NONE",RIGHT:"NONE",BOTTOM:"NONE"},Items:[]},target:-824633692101L}
[12:33:17] [Render thread/INFO] [STDOUT/]: [dev.latvian.kubejs.util.MapJS:of:32]: net.minecraft.nbt.CompoundNBT
[12:33:17] [Render thread/INFO] [STDOUT/]: [com.kotakotik.ponderjs.util.SceneBuilderJS$WorldInstructionsJS:lambda$mapJsConsumerToNBT$0:135]: new mapJsConsumerToNBT
[12:33:17] [Render thread/INFO] [STDOUT/]: [dev.latvian.kubejs.util.MapJS:of:31]: {output:{FluidName:"minecraft:empty",FacingModes:{LEFT:"ENABLED",TOP:"ENABLED",BACK:"ENABLED",FRONT:"ENABLED",RIGHT:"ENABLED",BOTTOM:"ENABLED"},Amount:0},miningDepth:3,redstoneManager:{LastState:0b,Name:"ignore"},catalyst:{Size:1,FacingModes:{LEFT:"NONE",TOP:"NONE",BACK:"NONE",FRONT:"NONE",RIGHT:"NONE",BOTTOM:"NONE"},Items:[]},work:{MaxProgress:20,Tick:0},energyStorage:{energy:0},x:3,y:3,tankBundle:{Input:{Size:1,Items:[]},Bar:{MaxProgress:10,Tick:0},Output:{Size:1,Items:[]}},z:3,id:"industrialforegoing:fluid_laser_base",augmentInventory:{Size:4,FacingModes:{LEFT:"NONE",TOP:"NONE",BACK:"NONE",FRONT:"NONE",RIGHT:"NONE",BOTTOM:"NONE"},Items:[]}}
[12:33:17] [Render thread/INFO] [STDOUT/]: [dev.latvian.kubejs.util.MapJS:of:32]: net.minecraft.nbt.CompoundNBT
```

as you can see it first tries to modify (or at least map) the nbt for the energy cells, and finally that of the tank itself, but all of them fail since its all `CompoundNBT` and `MapJS.nbt` only returns a `MapJS` instance or `null` 🤔

**Comment by kotakotik22** [comment]
pushed an attempt at a fix: 69579c2, ill put the preview jar here if my server can build it

**Comment by Quezler** [comment]
compiling a build with the kubejs version still at `1605.3.15-build.104` fixes both our latex & laser ponders, doesn't appear that the version upgrade is required for the compoundnbt fix to work 🙂
